src/core/defcomp.jl

```diff
@@ -195,10 +195,7 @@ macro defcomp(comp_name, ex)
             continue
         end
 
-        if @capture(elt, name_::datum_type_ = elt_type_(args__))
-            msg = "The following syntax has been deprecated in @defcomp: \"$name::$datum_type = $elt_type(...)\". Use curly bracket syntax instead: \"$name = $elt_type{$datum_type}(...)\""
-            Base.depwarn("$msg\n$(reduce((x,y) -> "$x\n$y", stacktrace()))", :defcomp)   
-        elseif ! @capture(elt, name_ = (elt_type_{datum_type_}(args__) | elt_type_(args__)))
+        if ! @capture(elt, name_ = (elt_type_{datum_type_}(args__) | elt_type_(args__)))
```

Comment:
@ckingdon95 does this look right I'm bad at macros

Reply:
Yea I was struggling with that for most of the deprecations to, I’m not sure where we would use a special error message and where we just delete the method so it ends up as a method error (like with the timestep indexing with integers).